components/Layout.vue

```diff
@@ -93,6 +102,34 @@ const section = navigation.find((s) =>
 );
 const { toc } = useContent();
 const showTocDropdown = ref(false);
+
+let visibleHeadingId = ref('');
+let observer = null;
+onMounted(() => {
+  const observerOptions = {
+    threshold: 0,
+    rootMargin: '-20% 0px -80% 0px',
```

Comment:
This means that when a section heading is 20% from the top it is considered active.
Note that this means on first page load, when the overview is mainly visible, there is no sections highlighted.
It also means a small section at the bottom may never become highlighted.